src/main/kotlin/fr/social/gouv/agora/usecase/qag/GetQagModeratingListUseCase.kt

```diff
@@ -17,26 +18,36 @@ class GetQagModeratingListUseCase(
     private val responseQagRepository: ResponseQagRepository,
     private val thematiqueRepository: ThematiqueRepository,
     private val supportRepository: GetSupportQagRepository,
+    private val qagModeratingLockRepository: QagModeratingLockRepository,
     private val mapper: QagModeratingMapper,
 ) {
 
     companion object {
-        private const val MAX_MODERATING_LIST_SIZE = 10
+        private const val MAX_MODERATING_LIST_SIZE = 5
     }
 
     fun getQagModeratingList(userId: String): List<QagModerating> {
         val thematiqueMap = mutableMapOf<String, Thematique?>()
 
         return qagInfoRepository.getAllQagInfo()
             .filter { qagInfo -> qagInfo.status == QagStatus.OPEN }
+            .filter { qagInfo ->
+                val lockingUserId = qagModeratingLockRepository.getUserIdForQagLocked(qagInfo.id)
+                lockingUserId == null || lockingUserId == userId
+            }
             .sortedBy { qagInfo -> qagInfo.date }
             .mapNotNullWhile(
                 transformMethod = { qagInfo -> toQagModerating(qagInfo, userId, thematiqueMap) },
                 loopWhileCondition = { qagPreviewList -> qagPreviewList.size < MAX_MODERATING_LIST_SIZE },
-            )
+            ).onEach { qagModerating ->
+                qagModeratingLockRepository.setQagLocked(
+                    qagId = qagModerating.id,
+                    userId = userId,
+                )
+            }
     }
 
-    fun getModeratingQagCount(): Int {
+    fun getModeratingQagCount(userId: String): Int {
```

Comment:
Tu peux l'enlever, il ne sert plus à rien

Reply:
you're right!